data/gnome-screensaver.1
.\" Copyright (C) 2007 Sven Arvidsson <sa@whiz.se>
.\"
.\" This is free software; you may redistribute it and/or modify
.\" it under the terms of the GNU General Public License as
.\" published by the Free Software Foundation; either version 2,
.\" or (at your option) any later version.
.\"
.\" This is distributed in the hope that it will be useful, but
.\" WITHOUT ANY WARRANTY; without even the implied warranty of
.\" MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
.\" GNU General Public License for more details.
.\"
.\"You should have received a copy of the GNU General Public License along
.\"with this program; if not, write to the Free Software Foundation, Inc.,
.\"51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
.TH gnome-screensaver 1 "2007\-09\-27" "GNOME"
.SH NAME
gnome-screensaver \- screen saver and locker
.SH SYNOPSIS
.B gnome-screensaver
.RI [ OPTIONS... ]
.SH DESCRIPTION
GNOME Screensaver is the default screen saver and locker in a GNOME
desktop. It is designed with simplicity and security in mind.
.SH OPTIONS
.TP
.B \-\-no\-daemon
Don't become a daemon
.TP
.B \-\-debug
Enable debugging code
.P
This program also accepts the standard GTK options.
.SH AUTHORS
.B gnome-screensaver
is written by William Jon McCann <mccann@jhu.edu>.
.P
This manual page was written by Sven Arvidsson <sa@whiz.se>.
.SH SEE ALSO
.BR "gtk-options" (7)



NEWS
===============
Version 2.25.2
===============

- Fix pixmap leak (Matthias Clasen)
- Fix dialog border width

===============
Version 2.25.1
===============

- Add bevel around dialog
- Fixed #444927, get idle time from xorg
- Fixed #492949, screensavers-personal-slideshow should not show hidden directories (Jörg)
- Fixed #530561, "Random" preview is always the last saver (Maxim Ermilov)
- Fixed #561855, Calls wrong slideshow binary if there is one installed in $PATH (Frederic Crozat)

Translations:
None

===============
Version 2.24.1
===============

- Fixed #555701, libgnome-desktop:gnome_bg_create_pixmap() leaks pixmaps and X clients (Scott Remnant)
- Fixed #451498, Preview of "Blank screen" is not consistent (Neil Patel)
- Fixed #552119, Add GTK and GNOME to screensaver desktop (William Jon McCann)
- Fixed #552523, Screensaver suddenly stopped working (William Jon McCann)
- Fixed #555254, gnome-screensaver-dialog NULL pointer crash (William Jon McCann)
- Fixed #555491, gnome-screensaver should release mouse/keyboard when failed to grab both of them (Joey Yu Zheng)
- Fixed #560456, Cleaning up GTK Includes in gnome-screensaver (Maxim Ermilov)

Translations:
- Updated ar: Djihed Afifi
- Updated ast: Andre Klapper, Astur
- Updated be@latin: Ihar Hračyška, Ihar Hrachyshka
- Updated pt_BR: Leonardo Ferreira Fontenelle, Vladimir Melo


===============
Version 2.24.0
===============

- Updated LINGUAS, sr@latin.po, sr@Latn: Arangel Angov
- Updated ar: Djihed Afifi, Khaled Hosny
- Updated bg: Alexander Shopov
- Updated bn_IN: Runa Bhattacharjee
- Updated ca: Gil Forcada
- Updated da: Kenneth Nielsen
- Updated de: Hendrik Richter
- Updated dz: Pema Geyleg, Dawa pemo
- Updated en_GB: Philip Withnall, David Lodge
- Updated et: Ivar Smolin
- Updated fr: Claude Paroz
- Updated ga: Seán de Búrca
- Updated hr: Robert Sedak, Ante Karamatić
- Updated hu: Gabor Kelemen
- Updated it: Luca Ferretti
- Updated kn: Shankar Prasad
- Updated ko: Changwoo Ryu
- Updated mk: Arangel Angov, Clytie Siddall <clytie@riverland.net.au>
- Updated ml: Praveen Arimbrathodiyil, പ്രവീണ്‍ അരിമ്പ്രത്തൊടിയില്‍
- Updated mr: Sandeep Shedmake
- Updated nl: Wouter Bolsterlee
- Updated pl: Wadim Dziedzic, wadim dziedzic
- Updated pt_BR: Vladimir Melo
- Updated ro: Mugurel Tudor, Mișu Moldovan
- Updated sr.po, sr@latin: Goran Rakić
- Updated ta: I. Felix
- Updated tr: Baris Cicek


===============
Version 2.23.90
===============

- use standard icon names (Matthias Clasen)
- Paint widget background color as face background color.
- Add/remove windows as necessary in response to RANDR 1.2 events
- Honor the disable_user_switching lockdown key.
- Fixed #547013, The dialog incorrectly tell that the numlock is on (William Jon McCann)

Translations:
- Updated LINGUAS, sr@latin.po, sr@Latn: Goran Rakic
- Updated ar: Khaled Hosny
- Updated cs: Petr Kovar
- Updated eu: Inaki Larranaga Murgoitio, Iñaki Larrañaga Murgoitio
- Updated fi: Ilkka Tuohela, Timo Jyrinki
- Updated gu: Daniel Nylander, Sweta Kothari
- Updated he: Yair Hershkovitz  <yairhr@gmail.com>, Yaniv Abir <yanivabir at gmail dot com>
- Updated ja: Takeshi AIHANA
- Updated lt: Gintautas Miliauskas
- Updated pt: Duarte Loreto
- Updated pt_BR: Vladimir Melo
- Updated sv: Daniel Nylander

Help Manual Translations:
None

===============
Version 2.23.3
===============

        * Fix build issue (Wouter Bolsterlee)
        * Add a simple frame around user icon
        * Add a new child widget to use for themes to draw on

===============
Version 2.23.2
===============

        * Fix gamma lost during fade
        * Show default desktop background behind lock dialog
        * Add reporting of inhibitors to --query command

===============
Version 2.22.2
===============

        * Add a 160 char limit to the message text box.

===============
Version 2.22.1
===============

       * Fix for bug 525785 – CVE-2008-0887

===============
Version 2.22.0
===============

  Translators

        - Updated oc: Yannig Marchegay
        - Updated ca: Gil Forcada
        - Updated eu: Inaki Larranaga Murgoitio, Djihed Afifi <djihed@gmail.com>
        - Updated as: Amitakhya Phukan
        - Updated ar: Artur Flinta, Djihed Afifi <djihed@gmail.com>
        - Updated cs: Petr Kovar, Clytie Siddall <clytie@riverland.net.au>
        - Updated et: Priit Laes  <plaes at svn dot gnome dot
        - Updated gl: Ignacio Casal Quinteiro
        - Updated es: Jorge Gonzalez, Djihed Afifi <djihed@gmail.com>
        - Updated bn_IN: Runa Bhattacharjee
        - Updated nl: Wouter Bolsterlee
        - Updated pt: Duarte Loreto
        - Updated nb: Kjartan Maraas
        - Updated tr: Baris Cicek
        - Updated ne: Pawan Chitrakar
        - Updated lt: Gintautas Miliauskas
        - Updated th: Theppitak Karoonboonyanan
        - Updated zh_TW: Taiwan
        - Updated en_GB: Philip Withnall
        - Updated be: Alexander Nyakhaychyk
        - Updated fr: Claude Paroz
        - Updated zh_HK: Hong Kong
        - Updated pt_BR: Leonardo Ferreira Fontenelle
        - Updated de: Andre Klapper
        - Updated hu: Gabor Kelemen
        - Updated fi: bug #518255, Ilkka Tuohela
        - Updated da: Kenneth Nielsen
        - Updated ja: Takeshi AIHANA
        - Updated he: Yair Hershkovitz  <yairhr@gmail.com>
        - Updated be@latin: Ihar Hrachyshka
        - Updated ko: Changwoo Ryu
        - Updated sv: Daniel Nylander
        - Updated mk: Arangel Angov, Ignacio Casal Quinteiro  <nacho.resa@gmail.com>, Clytie Siddall <clytie@riverland.net.au>
        - Updated sk: Marcel Telka
        - Updated pl: Artur Flinta
        - Updated uk: Maxim Dziumanenko
        - Updated mr: Rahul Bhalerao, mr

===============
Version 2.21.6
===============

        * New theme transfer progress dialog. (John Millikin)
        * Migrate gnome-vfs to GIO (John Millikin)
        * Set default action after cancelling note.

===============
Version 2.20.0
===============

  gnome-screensaver


src/gnome-screensaver.c
/* -*- Mode: C; indent-tabs-mode: nil; c-basic-offset: 8; tab-width: 8 -*-
 *
 * Copyright (C) 2004-2006 William Jon McCann <mccann@jhu.edu>
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful, but
 * WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
 * 02111-1307, USA.
 *
 * Authors: William Jon McCann <mccann@jhu.edu>
 *
 */

#include "config.h"
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <errno.h>
#include <sys/time.h>
#include <sys/types.h>
#include <sys/wait.h>

#include <glib/gi18n.h>
#include <gtk/gtk.h>

#include "gnome-screensaver.h"
#include "gs-monitor.h"
#include "gs-debug.h"

void
gnome_screensaver_quit (void)
{
        gtk_main_quit ();
}

int
main (int    argc,
      char **argv)
{
        GSMonitor          *monitor;
        GError             *error = NULL;
        static gboolean     show_version = FALSE;
        static gboolean     no_daemon    = FALSE;
        static gboolean     debug        = FALSE;
        static GOptionEntry entries []   = {
                { "version", 0, 0, G_OPTION_ARG_NONE, &show_version, N_("Version of this application"), NULL },
                { "no-daemon", 0, 0, G_OPTION_ARG_NONE, &no_daemon, N_("Don't become a daemon"), NULL },
                { "debug", 0, 0, G_OPTION_ARG_NONE, &debug, N_("Enable debugging code"), NULL },
                { NULL }
        };

#ifdef ENABLE_NLS
        bindtextdomain (GETTEXT_PACKAGE, GNOMELOCALEDIR);
# ifdef HAVE_BIND_TEXTDOMAIN_CODESET
        bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
# endif
        textdomain (GETTEXT_PACKAGE);
#endif

        if (! gtk_init_with_args (&argc, &argv, NULL, entries, NULL, &error)) {
                if (error) {
                        g_warning ("%s", error->message);
                        g_error_free (error);
                } else {
                        g_warning ("Unable to initialize GTK+");
                }
                exit (1);
        }

        if (show_version) {
                g_print ("%s %s\n", argv [0], VERSION);
                exit (1);
        }

        /* debug to a file if in deamon mode */
        gs_debug_init (debug, ! no_daemon);
        gs_debug ("initializing gnome-screensaver %s", VERSION);

        monitor = gs_monitor_new ();

        if (monitor == NULL) {
                exit (1);
        }

        error = NULL;
        if (! gs_monitor_start (monitor, &error)) {
                if (error) {
                        g_warning ("%s", error->message);
                        g_error_free (error);
                } else {
                        g_warning ("Unable to start screensaver");
                }
                exit (1);
        }

        /* Don't close stdout and stderr for now */
        if (! no_daemon && daemon (0, 1)) {
                g_error ("Could not daemonize: %s", g_strerror (errno));
        }

        gtk_main ();

        g_object_unref (monitor);

        gs_debug ("gnome-screensaver finished");

        gs_debug_shutdown ();

        return 0;
}
